Preprocessing/adcpBinMappingPP.m

```diff
@@ -81,16 +81,32 @@
     % while beams 2 and 3 get further away. When roll is positive beam 3 is
     % closer to the surface while beam 2 gets further away.
     %
+
+    isUpwardLooking = true;
+
     distAlongBeams = sample_data{k}.dimensions{distAlongBeamsIdx}.data';
     if all(diff(distAlongBeams)<0)
         %invert distAlongBeams so we have increasing values
         % this is required for the interpolation function.
         distAlongBeams = distAlongBeams*-1;
+        isUpwardLooking = false;
     end
     pitch = sample_data{k}.variables{pitchIdx}.data * pi / 180;
     roll = sample_data{k}.variables{rollIdx}.data * pi / 180;
     beamAngle = sample_data{k}.meta.beam_angle * pi / 180;
 
+    %adjust pitch
+    info = sample_data{k}.meta.adcp_info;
+    pitch_bit = info.sensors_settings.Pitch;
```

Comment:
I suspect this line and below are only going to work for RDI ADCPs and will break for Norteks. If it is tru that the current PITCH variable is not the best one and that instead we should consider the adjusted pitch from the gimbal then it should be fixed somewhere else when assigning the content of the PITCH variable. What does Bec think about this? Is there a similar situation with Nortek?

Reply:
Comment from the sidelines of someone who is not a Nortek export :)

From nortek support group comment [here](https://support.nortekgroup.com/hc/en-us/articles/360029499692-Does-the-Signature-series-have-full-3D-compass-and-tilt-readings-) and oceannetworks processing comments [here](https://wiki.oceannetworks.ca/display/DP/49) I had assumed that RDI gimballed pitch correction does not apply to Norteks.

And more a style comment - I sometime get lost in long functions with lots of maths :)  A long time ago when playing around with this code I had started structuring as
 
```matlab
if isRDI% RDI 4 beams
% handle 4 beams

% TODO: 5th beam

elseif isNortek
   if number_of_beams == 3
   % current 3 beam code
   else
   % new 4-beam code
    end
end
```

So maybe after this patch this sort of thing can be done. And then maybe break out the ```if instrument_type``` parts to say Preprocessing/+TeledyneADCP/calculate_non_binmapped_heights.m and Preprocessing/+Nortek/calculate_non_binmapped_heights.m. It may end up the same transformation for 4-beam Nortek as an RDI, haven't looked at it for so long couldn't even tell you if the beam numbering is the same let alone its relation to pitch/roll. Just would prefer more splitting out.

This is purely selfish as I know one day someone is going to through more complex [Rowe Technologies](https://rowetechinc.com) adcp data at me.

Reply:
I suspect this is just a RDI correction. I don't know about Norteks, but I suggest changing it to only apply to RDI data and sort out the Norteks separately.

Reply:
@ggalibert  is correct here - The line mentioned only works for RDI. The "sensor_settings" fieldname is only available for RDIs, and this key-value lookup was implemented when doing the beam2earth functionality.